src/api/libcellml/generatorprofile.h

```diff
@@ -2509,14 +2509,40 @@ class LIBCELLML_EXPORT GeneratorProfile
      * @brief Set the @c std::string for the data structure for the variable
      * type object.
      *
-     * Set the @c std::string for the data structure for the variable type
-     * object.
+     * Set the @c std::string for the data structure for the variable
+     * type object. To be useful, the string should contain the <OPTIONAL_TYPE>
+     * tag, which will be replaced with a type for the external variable type,
```

Comment:
This is something that you, @hsorby and I quickly discussed at some point. You can currently configure a profile however you like, which means that it can result in generated code that is nonsensical, and there is nothing we can do about it.

However, as much as possible, our `Generator` will check whether something is indeed an empty string and, if so, will skip some code generation. This is what is done in `Generator::GeneratorImpl::addVariableTypeObjectCode()` for the data structure for the variable type object. IOW, we try to limit nonsensical code generation, but there is just so much we can do.